src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Extensions/NonPortable/CustomAttributeSearcher.cs

```diff
@@ -74,6 +74,30 @@ public IEnumerable<CustomAttributeData> GetMatchingCustomAttributes(E element, T
                         return true;
                     };
             }
+            else if (optionalAttributeTypeFilter.IsGenericTypeDefinition)
```

Comment:
Question: elsewhere in this file there are comments about some reflection metadata (e. g. IsSealed) not being present. Might that be true for the generics-related properties as well? If so, how should we handle that case?

Reply:
@buyaa-n I'm not sure I follow.

`optionalAttributeTypeFilter.IsSealed` is accessed inside a try-catch and is "optional" in the sense that it is just used as an optimization so we get the exception it's not a problem.

In contrast, if we sometimes can't check whether we're a generic type definition, then what do we do if the user passes an open generic type?

Reply:
Sorry, I misunderstood your initial comment 😝, IDK if the same issue could happen for `optionalAttributeTypeFilter.IsGenericTypeDefinition` @jkotas might help with that

Reply:
`IsGenericTypeDefinition` and other similar bool properties on System.Type won't throw `MissingMetadataException`.